tests/integration/erc20_integration_test.go

```diff
@@ -0,0 +1,186 @@
+package integration
+
+import (
+	"math/big"
+
+	"cosmossdk.io/math"
+	"github.com/cosmos/evm/testutil/integration/os/factory"
+	evmtypes "github.com/cosmos/evm/x/vm/types"
+	"github.com/ethereum/go-ethereum/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/evm/contracts"
+	commonfactory "github.com/cosmos/evm/testutil/integration/common/factory"
+	erc20types "github.com/cosmos/evm/x/erc20/types"
+	integrationutils "github.com/realiotech/realio-network/testutil/integration/utils"
+)
+
+var (
+	mintAmount     int64 = 1_000_000
+	transferAmount int64 = 1000
+)
+
+func (suite *EVMTestSuite) TestERC20RegisterAndConverting() {
```

Comment:
This is a very long tests and hard to understand what its trying to test. Either divide it into two tests or split it into more clear functions to help readability

Reply:
updated, wrap some assert func and helpers